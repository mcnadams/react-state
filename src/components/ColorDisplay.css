.colorDisplay {

  div:nth-child(6n) {
    animation: squishX 2s ease-in 3s infinite alternate;
  }
  
  div:nth-child(6n + 1) {
    animation: squishY 1s ease-in-out 1s infinite alternate;
  }
  
  div:nth-child(6n + 2) {
    animation: rotate 1s ease-in 1s infinite alternate;
  }

  div:nth-child(6n + 3) {
    animation: squishBoth 2s ease-in 1s infinite alternate;
  }
  
  div:nth-child(6n + 4) {
    animation: rotates 1s ease-in-out 1s infinite alternate;
  }
  
  div:nth-child(6n + 5) {
    animation: squishBoth 1s ease-in 1s infinite alternate;
  }

  @keyframes squishX {
    from { transform: scaleX(0.1) rotate(-180deg);}
    to   { transform: scaleX(1); }
  }
  
  @keyframes squishY {
    from { transform: scaleY(0.1) scaleX(0.8) rotate(20deg); }
    to   { transform: scaleY(1) scaleX(1); }
  }
  
  @keyframes squishBoth {
    from { transform: scaleY(0.1) scaleX(0.1); }
    to   { transform: scaleY(1) scaleX(1); }
  }
  
  @keyframes rotate {
    from { transform: scaleY(0.2) scaleX(0.5) rotate(180deg); }
    to   { transform: scaleY(1) scaleX(1); }
  }
  
  @keyframes rotates {
    from { transform: scaleY(0.9) scaleX(0.9) rotate(270deg); }
    to   { transform: scaleY(1) scaleX(1); }
  }
  
}