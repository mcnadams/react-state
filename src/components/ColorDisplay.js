import React from 'react';
import PropTypes from 'prop-types';
import styles from './Color.css';

function ColorDisplay({ color }) {
  return (
    <div id={styles['color-display']} className={styles[color]}></div>
  );
}

ColorDisplay.propTypes = {
  color: PropTypes.string.isRequired
};

export default ColorDisplay;
