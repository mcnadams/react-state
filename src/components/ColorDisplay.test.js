import React from 'react';
import { shallow } from 'enzyme';
import ColorDisplay from './ColorDisplay';

describe('Color Display', () => {

  it('makes a ColorDisplay', () => {
    const wrapper = shallow(<ColorDisplay color='red' />);
    expect(wrapper).toMatchSnapshot();
  });
});
