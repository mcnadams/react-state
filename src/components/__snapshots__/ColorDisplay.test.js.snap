// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color Display makes a ColorDisplay 1`] = `
<div
  className="red"
  id="color-display"
/>
`;
